src/Adapter/Carrier/Repository/CarrierRangeRepository.php

```diff
@@ -0,0 +1,213 @@
+                            'delimiter1' => $range->getFrom(),
+                            'delimiter2' => $range->getTo(),
+                        ]
+                    );
+                    $rangeId = $this->connection->lastInsertId();
+                    $rangesIds[$range->getFrom() . '-' . $range->getTo()] = $rangeId;
+                } else {
+                    $rangeId = $rangesIds[$range->getFrom() . '-' . $range->getTo()];
+                }
+
+                // Insert price in delivery table
+                $this->connection->insert(
+                    $this->dbPrefix . 'delivery',
+                    [
+                        'id_carrier' => $carrierId->getValue(),
+                        'id_' . $rangeTable => $rangeId,
+                        'id_zone' => $idZone,
+                        'price' => $range->getPrice(),
+                    ]
+                );
+            }
+        }
+
+        // Commit transaction
+        $this->connection->commit();
+    }
+
+    private function reset(CarrierId $carrierId, ShopConstraint $shopConstraint): void
+    {
+        $this->assertShopConstraint($shopConstraint);
+        $tablesToReset = [
+            'range_weight',
+            'range_price',
+            'delivery',
+        ];
+
+        foreach ($tablesToReset as $table) {
+            $this->connection->delete(
+                $this->dbPrefix . $table,
+                ['id_carrier' => $carrierId->getValue()]
+            );
+        }
+    }
+
+    private function getRangeMethodTable(int $calculatingMethod): string
+    {
+        switch ($calculatingMethod) {
+            case ShippingMethod::BY_WEIGHT:
+                return 'range_weight';
+            case ShippingMethod::BY_PRICE:
+                return 'range_price';
+            default:
+                throw new CarrierException(sprintf('Unknown shipping method "%s"', $calculatingMethod));
+        }
+    }
+
+    private function applyRangeTypeForQuery(QueryBuilder $queryBuilder, Carrier $carrier): QueryBuilder
+    {
+        // Define which table to join based on carrier shipping method
+        $tableRange = $this->getRangeMethodTable($carrier->shipping_method);
+
+        // Join the range table and order by range
+        $queryBuilder->innerJoin(
+            'cd',
+            $this->dbPrefix . $tableRange,
+            'cr',
+            'cd.id_' . $tableRange . ' = cr.id_' . $tableRange . ' AND cr.id_carrier = :carrierId'
+        )
+            ->orderBy('cr.id_' . $tableRange, 'ASC');
+
+        return $queryBuilder;
+    }
+
+    private function applyShopConstraint(QueryBuilder $queryBuilder, ShopConstraint $shopConstraint): QueryBuilder
+    {
+        // We need temporary to groupby by zone, range and price to avoid duplicates for retrocompatibility
+        $queryBuilder->groupBy('cd.id_zone, cr.delimiter1, cr.delimiter2, cd.price');
+
+        // todo: make shop constraint magic here!
```

Comment:
You need to handle the all shops condition at least `WHERE id_shop = NULL AND id_shop_group = NULL`